tools/generate-authors-page

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+# Fetch and generate /authors page.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from typing import Text, Union, Optional, Dict, List, Any
+
+import sys
+import os
+import argparse
+from datetime import date
+import requests
+import simplejson as json
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
+os.environ['DJANGO_SETTINGS_MODULE'] = 'zproject.settings'
+from django.conf import settings
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
+from zerver.lib.utils import split_by
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--max-retries', metavar='DIR', type=int, default=2,
+                    help='Number of times to retry fetching data from Github')
+args = parser.parse_args()
+
+def fetch_data():
+    # type: () -> Optional[List[Dict[str, Any]]]
+    r = requests.get('https://api.github.com/repos/zulip/zulip/'
+                     'stats/contributors')  # type: requests.Response
+
+    if r.status_code == 200:
+        return r.json()
+    return None
+
+def gen_authors_row(people):
+    # type: (List[Optional[Dict[str, Union[Text, int]]]]) -> Text
+    md = "<div class='span6'><table class='authors_row'><tr>"  # type: Text
+
+    for person in people:
+        md += ("<td><a href='https://github.com/{name}'>"
+               "<div class='avatar float-left'>"
+               "<img class='avatar_img' src='{avatar}'>"
+               "</div>"
+               "<div class='info float-right'>"
+               "<b>@{name}</b><br />{commits} commits"
+               "</div>"
+               "</a></td>").format(
+            avatar=person.get("avatar"),
+            name=person.get("name"),
+            commits=person.get("commits")
+        ) if person else "<td></td>"
+
+    md += "</tr></table></div>"
+    return md
+
+
+while args.max_retries:
+    json_data = fetch_data()  # type: Optional[List[Dict[str, Any]]]
+    if json_data:
```

Comment:
It will http://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module